### PR TITLE
Add Possum Labs (HLP Portal)

### PR DESCRIPTION
**Pull request by PossumLabsCrypto**
##### Name (to be shown on DefiLlama): 
Possum Labs

##### Twitter Link: 
https://twitter.com/Possum_Labs

##### List of audit links if any: 
https://possum-labs.gitbook.io/docs/security/audits

##### Website Link: 
https://www.possumlabs.io/

##### Logo (High resolution, will be shown with rounded borders):
https://assets.coingecko.com/coins/images/33596/standard/PSM.png?1702521046

##### Current TVL:
 ca. 145000 USD

##### Treasury Addresses (if the protocol has treasury)
0xAb845D09933f52af5642FC87Dd8FBbf553fd7B33

##### Chain:
Arbitrum

##### Coingecko ID (so your TVL can appear on Coingecko, leave empty if not listed): (https://api.coingecko.com/api/v3/coins/list)
possum 

##### Short Description (to be shown on DefiLlama):
Possum Labs is on the mission to spearhead the development of positive sum DeFi.

The first protocol released by Possum Labs is called Portals and enables users to receive upfront fixed yield on their staked tokens without any protocol fees.

Our second contract is the Time Rift which is a new primitive for community onboarding. A positive-sum alternative to airdrops.

For more information, visit our documentation: https://possum-labs.gitbook.io/docs/

##### Token address and ticker if any:
0x17a8541b82bf67e10b0874284b4ae66858cb1fd5
PSM

##### Category (full list at https://defillama.com/categories) *Please choose only one:
Yield

##### methodology (what is being counted as tvl, how is tvl being calculated):
Staked tokens in our protocol. TVL = tokens * price

##### Github org/user (Optional, if your code is open source, we can track activity):
PossumLabsCrypto

### CODE REVIEW

**Comment by llamatester** [comment]
Error while running adapter at projects/PossumLabs: 
        
 
  
    
    
    Error: 
            Incorrect time format for the hallmark: [155591158, HLP Portal Launch] ,please use unix timestamp
            
    Error saving cache: Error: ENOENT: no such file or directory, open '/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/local_cache/zlib-1.0/BlockCache.json'
        at Object.openSync (node:fs:596:3)
        at Object.writeFileSync (node:fs:2322:35)
        at process.saveCache (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/cache.js:138:17)
        at process.emit (node:events:529:35)
        at process.exit (node:internal/process/per_thread:191:15)
        at process.handleError (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/utils/handleError.js:18:11)
        at process.emit (node:events:517:28)
        at emit (node:internal/process/promises:149:20)
        at processPromiseRejections (node:internal/process/promises:283:27)
        at process.processTicksAndRejections (node:internal/process/task_queues:96:32) {
      errno: -2,
      syscall: 'open',
      code: 'ENOENT',
      path: '/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/local_cache/zlib-1.0/BlockCache.json'
    }

**Comment by llamatester** [comment]
Error while running adapter at projects/PossumLabs: 
        
 
  
    
    
    Error: Protocol doesn't have total tvl, make sure to export a tvl key either on the main object or in one of the chains
    Truncated error stack:
    
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:217:11

File: projects/PossumLabs/index.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    timetravel: true,
+    misrepresentedTokens: false,
+    methodology: 'Get the number of HLP tokens staked and HLP price to calculate USD TVL.',
+    start: 1701313010,
+    arbitrum: {
+        tvl,
+    }
+};
+
+module.exports = {
```
Comment (g1nt0ki) [inline projects/PossumLabs/index.js:36]: merge this with `module.exports` above, else it will replace it

File: projects/PossumLabs/index.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
+const HLP_Portal = '0x1f3Af095CDa17d63cad238358837321e95FC5915';
```
Comment (g1nt0ki) [inline projects/PossumLabs/index.js:1]: is this address correct?

File: projects/PossumLabs/index.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+// const TIME_RIFT = '0x6df4EF024089ab148078fdD88f5BF0Ee63248D3E';
+
+async function tvl(_, _1, _2, { api }) {
+
+    const stakeBalanceHLP = await api.call({
+        abi: 'uint256:totalPrincipalStaked',
+        target: HLP_Portal,
+    });
+
+    const PrincipalPrice = await api.call({
```
Comment (g1nt0ki) [inline projects/PossumLabs/index.js:12]: dont think you need to fetch HLP price, think we have support for it

**Comment by llamatester** [comment]
Error while running adapter at projects/PossumLabs: 
        
 
  
    
    
    Error: Protocol doesn't have total tvl, make sure to export a tvl key either on the main object or in one of the chains
    Truncated error stack:
    
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:217:11

**Comment by llamatester** [comment]
The adapter at projects/PossumLabs exports TVL: 
        
 
 
    arbitrum                  0
    
    total                    0

**Comment by llamatester** [comment]
The adapter at projects/PossumLabs exports TVL: 
        
 
 
    arbitrum                  0
    
    total                    0

**Comment by llamatester** [comment]
The adapter at projects/PossumLabs exports TVL: 
        
 
 
    arbitrum                  0
    
    total                    0

**Comment by PossumLabsCrypto** [comment]
Thank you for the help! @g1nt0ki 

I don´t understand why it returns TVL 0 now.

Any suggestion?

This is the tracker parameter of staked HLP in the contract:
https://arbiscan.io/address/0x24b7d3034C711497c81ed5f70BEE2280907Ea1Fa#readContract#F27

**Comment by llamatester** [comment]
The adapter at projects/PossumLabs exports TVL: 
        
 
 
    arbitrum                  144.26 k
    
    total                    144.26 k

**Comment by g1nt0ki** [comment]
@PossumLabsCrypto thanks for the PR

**Comment by llamatester** [comment]
The adapter at projects/possumLabs exports TVL: 
        
 
 
    arbitrum                  145.89 k
    
    total                    145.89 k

**Pull request merged**